api/BUILD

```diff
@@ -12,11 +12,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+
 package(default_visibility = ["//visibility:public"])
 
+bool_flag(
+    name = "with_abseil",
+    build_setting_default = False,
+)
+
 cc_library(
     name = "api",
     hdrs = glob(["include/**/*.h"]),
+    defines = select({
+        ":with_external_abseil": ["HAVE_ABSEIL"],
+        "//conditions:default": [],
+    }),
     strip_include_prefix = "include",
     tags = ["api"],
+    deps = select({
+        ":with_external_abseil": [
+            "@com_google_absl//absl/base",
+            "@com_google_absl//absl/types:any",
+            "@com_google_absl//absl/types:variant",
+            "@com_google_absl//absl/types:span",
```

Comment:
nit - do we need to add a dependency for absl::span?

Reply:
`@com_google_absl//absl/types:any` and `@com_google_absl//absl/types:span` are removed. The failed jobs seem not be the problem of this PR.